### PR TITLE
Add standards for Go programming language

### PR DESCRIPTION
**Pull request by cfarrend**
Currently the @qctrl/devops-engineering team are the only team that are making use of [Go](https://golang.org/) in their repositories, however we encourage anyone to make a contribution to our source code and wish to establish some standards. The Style, Docstring, Testing and Linting standards are all from https://golang.org

### CODE REVIEW

**Comment by charmasaur** [comment]
You can tell a language has done it right when all the standards come from official pages. LGTM but I'll let Rob approve.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -130,13 +130,14 @@ then, you MUST move to [First-party](#first-party).
...
-| Python     | [PEP 8](https://www.python.org/dev/peps/pep-0008/) | [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) | [pytest](https://pytest.org/)                              | [Pylint](https://www.pylint.org/)                                          |
+| Language   | Style                                                    | Docstrings                                                        | Testing                                                    | Linting                                                                    |
+|------------|----------------------------------------------------------|-------------------------------------------------------------------|------------------------------------------------------------|----------------------------------------------------------------------------|
+| Go         | [Effective Go](https://golang.org/doc/effective_go.html) | [GoDoc](https://blog.golang.org/godoc)                            | [go test](https://golang.org/pkg/testing/)                 | [golint](https://github.com/golang/lint)                                   |
```
Comment (robertlove) [inline CONTRIBUTING.md:135]: "GoDoc" should be "Godoc". If you don't believe me, see https://blog.golang.org/godoc

Comment (robertlove) [inline CONTRIBUTING.md:135]: "golint" should be "Golint". If you don't believe me, see https://github.com/golang/lint

Comment (cfarrend) [reply CONTRIBUTING.md:135]: I was confused about this one, I initially thought the same thing until I saw this from the GoDoc website https://godoc.org/-/about. Perhaps there is a difference between the docstring style and the site that hosts the documentation itself?

Comment (cfarrend) [reply CONTRIBUTING.md]: Updated

Comment (cfarrend) [reply CONTRIBUTING.md]: I have re-read the about page, this is a good quote: "The guidelines for writing documentation for the godoc tool apply to GoDoc."

**Comment by cfarrend** [comment]
Hey @qctrl/support anything I'm missing from this?

**Review by charmasaur — APPROVED** [review]
LGTM, if anybody has more comments they can always be addressed in a follow-up PR

**Pull request merged**